circuits/core/manager.py

```diff
@@ -948,7 +959,7 @@ def run(self, socket=None):
         self.fire(started(self))
 
         try:
-            while self.running or self._queue:
+            while self.running or len(self):
```

Comment:
I think we used this explicit check because if a component would overwrite `__len__` this would have implications. See https://github.com/circuits/circuits/issues/51 for a similar issue.